charts/gardener-metrics-exporter/charts/runtime/templates/serviceaccount.yaml

```diff
@@ -0,0 +1,27 @@
+{{- if not .Values.global.virtualGarden.enabled }}
+apiVersion: v1
```

Comment:
These two service account templates are the same. Do we need this logic?

Reply:
But if they are the same why do we need an if statement at all? OK maybe if we change the service account for the virtual cluster and then it is different but that is not the case now.

Reply:
The serviceaccount template in the `application` section will only be deployed if virtualGarden is enabled and there is no `user.name` meaning that we will want to get its token for auth.

In the `runtime` section the serviceaccount is deployed if virtualGarden is not enabled
or 
virtualGarden is enabled and we have user.name set and projected serviceaccount tokens are also enabled.

We cannot avoid the `if` statement because sometimes we do not want `serviceaccount` deployed at all. For example when we have deployed it in the virtual garden or when we are going to use certificate for auth. Then we will default to the `default` serviceaccount.